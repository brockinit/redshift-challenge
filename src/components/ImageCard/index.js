const ImageCard = props => {
  return (
    <div className="grid-container-outer">
      <div
        className="grid-image-block"
        style={{ backgroundImage: `url(${props.link})` }}
      />
    </div>
  );
};

export default ImageCard;
