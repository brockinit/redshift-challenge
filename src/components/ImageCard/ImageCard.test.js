import {
  renderToSnapshot,
  isClassVNode,
  isFunctionalVNode,
} from 'inferno-test-utils';
import ImageCard from './';

test('ImageCard should be a functional component', () => {
  const isClass = isClassVNode(<ImageCard />);
  const isFunctional = isFunctionalVNode(<ImageCard />);
  expect(isClass).toBe(false);
  expect(isFunctional).toBe(true);
});

test('should have an image prop', () => {
  const shallowRender = renderToSnapshot(<ImageCard />);
  expect(shallowRender).toMatchSnapshot();
});
