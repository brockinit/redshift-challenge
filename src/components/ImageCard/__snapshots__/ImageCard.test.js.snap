// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCard should render correctly 1`] = `
<div
  className="grid-container-outer"
>
  <div
    className="grid-image-block"
    style={
      Object {
        "backgroundImage": "url(undefined)",
      }
    }
  >
    <div
      className="grid-hover"
    />
  </div>
</div>
`;
